# Metnum Kelompok 3 TIF-E

### Anggota:
1. Alfred Josafat Elyazar 215150207111046
2. Sulthan Abiyyu Hakim 215150201111011
3. Aditya Rizky Ramadhan 215150201111007
4. Muhammad Ubaidillah 215150201111010

## Latihan 1.
- Jalankan program berikut, perhatikan hasilnya.
- Ubah program berikut dalam bentuk comprehension dengan hasil yang sama 

In [1]:
y = []
for i in range(10):
    for j in range(10):
        if i == j:
            y.append(i+j)
y

[0, 2, 4, 6, 8, 10, 12, 14, 16, 18]

In [2]:
y = [i+j for i in range(10) for j in range(10) if i == j]
y

[0, 2, 4, 6, 8, 10, 12, 14, 16, 18]

# Latihan 2.
Write a function my_max(x) to return the maximum (largest) value in x. Don’t use the built-in Python function max.

In [3]:
def my_max(x):
  max = x[0]
  for i in x:
    if i > max:
      max = i
  return max

  
x = [10,1,3,9,20,21]
max = my_max(x)
max

21

# Latihan 3
Write a function my_n_max(x, n) to return a list consisting of the n largest elements of x. You may use the my_max function you have created . You may also assume that x is a one-dimensional list with no duplicate entries, and that n is strictly positive integer smaller than the length of x



In [15]:
x = [7, 9, 10, 5, 8, 3, 4, 6, 2, 1]

def my_n_max(x, n):
  out = []
  for i in range(n):
    out.append(my_max(x))
    x.remove(my_max(x))
  return out

  return out

x = [7, 9, 10, 5, 8, 3, 4, 6, 2, 1]
n = 3
out = my_n_max(x, n)
print(out)

[10, 9, 8]


Jika diberikan data sebagai berikut :

x = [7, 9, 10, 5, 8, 3, 4, 6, 2, 1]

n = 3

maka hasilnya :

out = my_n_max(x, n)

print(out)

[10, 9, 8]


# Latihan 4
Let m be a matrix of positive integers. Write a function my_trig_odd_even(m) to return an array q, where q[i, j] = sin (m[i, j]) if m[i, j] is even, and q[i, j] = cos (m[i, j]) if m[i, j] is odd.

In [13]:
import numpy as np

def my_trig_odd_even(m):
  q = []
  for i in range(len(m)):
    for j in range(len(m[0])):
      if m[i][j] % 2 == 0:
        q.append(np.sin(m[i][j]))
      else:
        q.append(np.cos(m[i][j]))
  return q

m = list([
    [1,2,3],
    [4,5,6]
    ])

q = my_trig_odd_even(m)
q

[0.5403023058681398,
 0.9092974268256817,
 -0.9899924966004454,
 -0.7568024953079282,
 0.28366218546322625,
 -0.27941549819892586]

Jika diberikan data sebagai berikut 

m = list([[1,2,3],[4,5,6]])

q = my_trig_odd_even(m)

q

maka hasilnya 

[0.5403023058681398,
 0.9092974268256817,
 -0.9899924966004454,
 -0.7568024953079282,
 0.28366218546322625,
 -0.27941549819892586]



#Latihan 5
Write a function with my_find(M), where output is a list of indices i where M[i] is 1. You may assume that M is a list of only ones and zeros. Do not use the built-in Python function find.



Jika diberikan 

M = [1,0,1,1,0,0,1]

my_find(M)

Keluarannya :

[0, 2, 3, 6]

In [14]:
def my_find(M):
  indices = []
  for idx, value in enumerate(M):
    if value == 1:
      indices.append(idx)
  
  return indices

M =  [1,0,1,1,0,0,1]
print(my_find(M))

[0, 2, 3, 6]
